### PR TITLE
Introduce a `Class` map

### PR DESCRIPTION
**Pull request by jedel1043**
Related to #3314.

This change improves the ergonomics of the `Class` API by adding a class map that can be used to fetch the constructor and prototype of a registered class. This also slightly improves its API by implementing some missing features like using symbols as method names.

### CODE REVIEW

File: boa_engine/src/context/mod.rs [hunk 1]
```diff
@@ -330,28 +332,48 @@ impl<'host> Context<'host> {
     ///    // ...
     /// }
     ///
-    /// context.register_global_class::<MyClass>();
+    /// context.register_global_class::<MyClass>()?;
     /// ```
-    pub fn register_global_class<T>(&mut self) -> JsResult<()>
+    pub fn register_global_class<C>(&mut self) -> JsResult<()>
```
Comment (HalidOdat) [inline boa_engine/src/context/mod.rs:337]: Maybe it would be good to implement ability to remove registered classes?

**Review by HalidOdat — APPROVED** [review]
Looks great! Just some comments, they don't block merging though :)

File: boa_engine/src/realm.rs [hunk 2]
```diff
@@ -102,6 +111,25 @@ impl Realm {
...
+    pub fn has_class<C: Class>(&self) -> bool {
+        self.inner
+            .host_classes
+            .borrow()
+            .contains_key(&TypeId::of::<C>())
+    }
+
+    /// Gets the constructor and prototype of the class `C` if it is registered in the class map.
+    #[must_use]
+    pub fn get_class<C: Class>(&self) -> Option<StandardConstructor> {
```
[the remark this answers is not in the document]
Comment (jedel1043) [reply boa_engine/src/realm.rs:125]: This cannot panic under any circumstance. The borrow doesn't escape the inner context, the returned value is a clone of the inner value and there are no ways to get a `GcRef/GcRefMut` from the API.

**Review by HalidOdat — APPROVED** [review]
Looks perfect to me! :)

**Review by raskad — APPROVED** [review]

**Pull request merged**